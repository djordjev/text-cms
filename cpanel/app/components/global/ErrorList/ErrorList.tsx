import { IconAlertCircle } from '@tabler/icons-react';
import classnames from 'classnames';
import { FC } from 'react';

export interface ErrorListProps {
  className?: string;
  errors?: string[];
}

const ErrorList: FC<ErrorListProps> = (props) => {
  const { className, errors } = props;

  // Styles
  const classes = classnames('u-alert u-alert-error u-mb-3x', className);

  // Markup
  const renderErrors = () => {
    if (!errors) return null;

    return errors.map((e) => <div key={e}>{e}</div>);
  };

  // Short-circuit
  if (!errors?.length) return null;

  return (
    <div role="alert" className={classes}>
      <IconAlertCircle />
      <div>{renderErrors()}</div>
    </div>
  );
};

export { ErrorList };
