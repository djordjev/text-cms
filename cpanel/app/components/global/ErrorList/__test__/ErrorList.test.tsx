import { createRemixStub } from '@remix-run/testing';
import { render, screen } from '@testing-library/react';

import { ErrorList, ErrorListProps } from '../ErrorList';

test('ErrorList', () => {
  const props: ErrorListProps = {
    errors: ['first error', 'second error']
  };

  const createComponent = () => {
    const RemixStub = createRemixStub([
      {
        path: '/',
        Component: () => <ErrorList {...props} />
      }
    ]);

    return <RemixStub />;
  };

  const { container, rerender } = render(createComponent());

  props.errors!.forEach((e) => expect(screen.getByText(e)).toBeInTheDocument());

  expect(container.childNodes).toMatchSnapshot();

  // does not render if no errors
  props.errors = undefined;
  rerender(createComponent());

  expect(container.childNodes).toHaveLength(0);
});
