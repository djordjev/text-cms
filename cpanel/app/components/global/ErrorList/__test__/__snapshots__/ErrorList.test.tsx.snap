// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorList 1`] = `
NodeList [
  <div
    class="u-alert u-alert-error u-mb-3x"
    role="alert"
  >
    <svg
      class="tabler-icon tabler-icon-alert-circle"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 12a9 9 0 1 0 18 0a9 9 0 0 0 -18 0"
      />
      <path
        d="M12 8v4"
      />
      <path
        d="M12 16h.01"
      />
    </svg>
    <div>
      <div>
        first error
      </div>
      <div>
        second error
      </div>
    </div>
  </div>,
]
`;
